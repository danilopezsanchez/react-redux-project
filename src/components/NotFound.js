import ghost from "../utils/images/ghost.gif";

const NotFound = () => {
	return (
		<div className="notfoundContainer">
			<h1>The page your are looking for doesn't exist</h1>
			<h2>Maybe the ghostbusters can help</h2>
			<img src={ghost} />
		</div>
	)
}

export default NotFound;