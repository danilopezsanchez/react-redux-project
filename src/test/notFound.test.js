import * as React from 'react';
import { render } from '@testing-library/react';
import NotFound from '../components/NotFound';


describe('NotFound', () => {
    it('matches the snapshot', () => {
        var component = render(
            <NotFound />
        );
        expect(component).toMatchSnapshot();
    });

	it('check have the h1', () => {
        var component = render(<NotFound />);

        var headTitle = component.getByRole('heading', { level: 1 })
        expect(headTitle).toBeInTheDocument();
    });

	it('check have the h2', () => {
        var component = render(<NotFound />);

        var headTitle = component.getByRole('heading', { level: 2 })
        expect(headTitle).toBeInTheDocument();
    });

	it('check have the img', () => {
        var component = render(<NotFound />);

        var img = component.getByRole('img')
        expect(img).toBeInTheDocument();
    });
});