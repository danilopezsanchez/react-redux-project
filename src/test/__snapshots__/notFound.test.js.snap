// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="notfoundContainer"
      >
        <h1>
          The page your are looking for doesn't exist
        </h1>
        <h2>
          Maybe the ghostbusters can help
        </h2>
        <img
          alt="gif page not found"
          src="ghost.gif"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="notfoundContainer"
    >
      <h1>
        The page your are looking for doesn't exist
      </h1>
      <h2>
        Maybe the ghostbusters can help
      </h2>
      <img
        alt="gif page not found"
        src="ghost.gif"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
